tools/benchmark/cmd/watch.go

```diff
@@ -163,14 +174,35 @@ func doWatch(stream etcdserverpb.Watch_WatchClient, requests <-chan etcdserverpb
 		results <- result{errStr: errStr, duration: time.Since(st)}
 		bar.Increment()
 	}
+	wg.Done()
+
+	<-putPhaseStartCh
+
 	for {
+		st := time.Now()
 		_, err := stream.Recv()
 		var errStr string
 		if err != nil {
 			errStr = err.Error()
 		}
-		results <- result{errStr: errStr}
+		results <- result{errStr: errStr, duration: time.Since(st)}
 		bar.Increment()
+
+		atomic.AddInt64(&completedWatchPutTotal, 1)
+		if atomic.LoadInt64(&completedWatchPutTotal) == int64(eventsTotal) {
+			break
+		}
 	}
+
 	wg.Done()
 }
+
+func doPutForWatch(ctx context.Context, client etcdserverpb.KVClient, requests <-chan etcdserverpb.PutRequest) {
```

Comment:
doPut() not only increases the bar, but also sends result and calls wg.Done(). Therefore using it in watch benchmark would be complicated.